NEWS.md

```diff
@@ -0,0 +1,302 @@
+  "a b c"
+  "d\ne\nf"
+  ```
+  # also it's an error to output a string containing a NUL when using NUL separator
+  $ jq -n --raw-output0 '"\u0000"'
+  jq: error (at <unknown>): Cannot dump a string containing NUL with --raw-output0 option
+- Fix assert crash and validate JSON for `--jsonarg`. @wader #2658
+- Remove deprecated `--argfile` option. @itchyny #2768
+
+## Language changes
+
+- Use decimal number literals to preserve precision. Comparison operations respects precision but arithmetic operations might truncate. #1752 @leonid-s-usov
+  ```sh
+  # precision is preserved
+  $ jq -n '100000000000000000'
+  100000000000000000
+  # comparison respects precision (this is false in JavaScript)
+  $ jq -n '100000000000000000 < 100000000000000001'
+  true
+  # arithmetic operations might truncate (same as JavaScript)
+  $ jq -n '100000000000000000+10'
+  100000000000000020
+  ```
+- Ads new builtin `pick(stream)` to emit a projection of the input object or array. @pkoppstein #2656
+  ```sh
+  $ jq -n '{"a": 1, "b": {"c": 2, "d": 3}, "e": 4} | pick(.a, .b.c, .x)'
+  {
+    "a": 1,
+    "b": {
+      "c": 2
+    },
+    "x": null
+  }
+  ```
+- Adds new builtin `debug(msgs)` that works like `debug` but applies a filter on the input before writing to stderr.
+  ```sh
+  $ jq -n '1 as $x | 2 | debug("Entering function foo with $x == \($x)", .) | (.+1)'
+  ["DEBUG:","Entering function foo with $x == 1"]
+  ["DEBUG:",2]
+  3
+  $ jq -n '{a: 1, b: 2, c: 3} | debug({a, b, sum: (.a+.b)})'
+  ["DEBUG:",{"a":1,"b":2,"sum":3}]
+  {
+    "a": 1,
+    "b": 2,
+    "c": 3
+  }
+  ```
+- Adds new builtin `scan($re; $flags)`. Was documented but not implemented. @itchyny #1961
+  ```sh
+  # look for pattern "ab" in "abAB" ignoring casing
+  $ jq -n '"abAB" | scan("ab"; "i")'
+  "ab"
+  "AB"
+  ```
+- Adds new builtin `abs` to get absolute value. This was previously possibly using `length` or `fabs` but naming was a bit confusing. @pkoppstein #2767
+- Allow `if` without `else`-branch. When skipped the `else`-branch will be `.` (identity). #1825 @chancez #2481 @wader
+  ```sh
+  # convert 1 to "one" otherwise keep as is
+  $ jq -n '1,2 | if . == 1 then "one" end'
+  "one"
+  2
+  # behaves the same as
+  $ jq -n '1,2 | if . == 1 then "one" else . end'
+  "one"
+  2
+  # also works with elif
+  $ jq -n '1,2,3 | if . == 1 then "one" elif . == 2 then "two" end
+  "one"
+  "two"
+  3
+  ```
+- Allow use of `$binding` as key in object literals. 8ea4a55 @nicowilliams
+  ```sh
+  $ jq -n '"a" as $key | {$key: 123}'
+  {
+    "a": 123
+  }
+  # previously parentheses were needed
+  $ jq -n '"a" as $key | {($key): 123}'
+  {
+    "a": 123
+  }
+  ```
+- Allow dot between chained indexes when using `.["index"]` #1168 @nicowilliams
+  ```sh
+  $ jq -n '{"a": {"b": 123}} | .a["b"]'
+  123
+  # now this works also
+  $ jq -n '{"a": {"b": 123}} | .a.["b"]'
+  123
+  ```
+- Speed up and refactor some builtins, also remove `scalars_or_empty/0`. #1845 @muhmuhten
```

Comment:
We should probably mention the try/catch bug fix as well.  That one had lots of dup issues.